semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -203,6 +203,14 @@ trait TextDocumentOps { self: SemanticdbOps =>
 
             saveSymbol(gsym)
             if (gsym.isClass && !gsym.isTrait) {
+              if (gsym.isAnonymousClass) {
+                val pos = gsym.pos.toMeta
```

Comment:
There is a “focus” method on compiler positions to make the start and end point to the same offset. That would avoid the need for startPosition below.

Reply:
Fixed